src/hammer-vlsi/synthesis/yosys/__init__.py

```diff
@@ -2,20 +2,34 @@
 # -*- coding: utf-8 -*-
 #
 # Yosys synthesis plugin for Hammer (as part of OpenROAD-flow installation).
+# mostly adoped from OpenLANE scripts found here: https://github.com/The-OpenROAD-Project/OpenLane/tree/master/scripts
 #
 # See LICENSE for licence details.
 
-import os
 from textwrap import dedent as dd
-from typing import List, Optional, Dict, Any
 
+# from hammer_utils import deepdict
+from hammer_vlsi.vendor import OpenROADTool, OpenROADSynthesisTool
+
+###########
+from hammer_vlsi import HammerTool, HammerToolStep, HammerToolHookAction, HierarchicalMode
+from hammer_vlsi.constraints import MMMCCorner, MMMCCornerType
+from hammer_utils import VerilogUtils, optional_map
+from hammer_vlsi import HammerSynthesisTool
 from hammer_logging import HammerVLSILogging
-from hammer_utils import deepdict
-from hammer_vlsi import HammerToolStep
-from hammer_vlsi.vendor import OpenROADSynthesisTool
+from hammer_vlsi import MMMCCornerType
+import hammer_tech
+
+from typing import Dict, List, Any, Optional
 
+import specialcells
+from specialcells import CellType, SpecialCell
+
+import os
+import json
+from collections import Counter
 
-class YosysSynth(OpenROADSynthesisTool):
+class YosysSynth(HammerSynthesisTool, OpenROADTool):
```

Comment:
Confused about why this is no longer mixing in `OpenROADSynthesisTool`. Are some things in that class conflicting with what you need for Yosys? Since Yosys is the OpenROAD synthesis tool... should some of the below things be moved there instead?